File: rust/src/models/invoice_custom_properties_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// Merge update of an invoice's custom property values (send a key with `null` to remove it).
/// Allowed at any status — custom properties stay editable after the invoice is finalized.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct InvoiceCustomPropertiesRequest {
    pub custom_properties: serde_json::Value,
}

impl InvoiceCustomPropertiesRequest {
    pub fn new(custom_properties: serde_json::Value) -> Self {
        Self { custom_properties }
    }
}

File: typescript/src/models/invoiceCustomPropertiesRequest.ts
// this file is @generated
/**
 * Merge update of an invoice's custom property values (send a key with `null` to remove it).
 * Allowed at any status — custom properties stay editable after the invoice is finalized.
 */
export interface InvoiceCustomPropertiesRequest {
  customProperties: any;
}

export const InvoiceCustomPropertiesRequestSerializer = {
  _fromJsonObject(object: any): InvoiceCustomPropertiesRequest {
    return {
      customProperties: object["custom_properties"],
    };
  },

  _toJsonObject(self: InvoiceCustomPropertiesRequest): any {
    return {
      custom_properties: self.customProperties,
    };
  },
};
